web3-react/src/libs/connections.ts

```diff
@@ -0,0 +1,126 @@
+import { Web3ReactHooks } from '@web3-react/core'
+import { Connector, AddEthereumChainParameter } from '@web3-react/types'
+import { buildInjectedConnector } from './injected'
+import { buildNetworkConnector } from './network'
+import { buildCoinbaseWalletConnector } from './coinbase'
+import { buildGnosisSafeConnector } from './gnosis'
+import { buildWalletConnectConnector } from './wallet-connect'
+import { CHAIN_INFO } from './constants'
+
+export interface Connection {
+  connector: Connector
+  hooks: Web3ReactHooks
+  type: ConnectionType
+}
+
+export enum ConnectionType {
+  COINBASE_WALLET = 'COINBASE_WALLET',
+  GNOSIS_SAFE = 'GNOSIS_SAFE',
+  INJECTED = 'INJECTED',
+  NETWORK = 'NETWORK',
+  WALLET_CONNECT = 'WALLET_CONNECT',
+}
+
+export function getIsInjected(): boolean {
+  return Boolean(window.ethereum)
+}
+
+export function getHasMetaMaskExtensionInstalled(): boolean {
+  return window.ethereum?.isMetaMask ?? false
```

Comment:
BTW, we need to get this into `web3-react` library.

Reply:
nit: this is going to be `true` for Brave browser too. Not sure if that matters, but you can look up my recent PR that changes this particular code in `interface` to not have false postiive results in Brave browser.

Reply:
I think it's important to ensure the example is not confusing. I think it's not that big of a hassle to add one additional conditional to check for Brave (`!window.ethereum?.isBraveWallet`) browser. If not, I would be happy to prefix this code block with a comment, so that we can teach users something new too!